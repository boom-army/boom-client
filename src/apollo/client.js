import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          users: offsetLimitPagination(),
        },
      },
    },
  }),
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
  },
  context: async ({ req }) => {
    const token = req?.headers?.authorization || null;
    const user= token ? JSON.parse(localStorage.getItem("user")) : null;
    return { user, isLoggedIn: !!token };
  },
});
