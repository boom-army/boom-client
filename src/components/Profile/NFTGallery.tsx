import Clear from "@mui/icons-material/Clear";
import DoNotDisturbOnIcon from "@mui/icons-material/DoNotDisturbOn";
import React, { useContext, useState, useMemo, useEffect } from "react";
import {
  Box,
  Button,
  Chip,
  IconButton,
  Link,
  Modal,
  Stack,
  Typography,
} from "@mui/material";
import { Loader } from "../Loader";
import {
  Metadata,
  MetadataData,
} from "@metaplex-foundation/mpl-token-metadata";
import { ThemeContext } from "../../contexts/theme";
import { currentCluster } from "../../utils/utils";
import { displayError } from "../../utils";
import { useConnection } from "@solana/wallet-adapter-react";
import {
  useEditProfileMutation,
  ProfileDocument,
  ProfileQuery,
} from "../../generated/graphql";
import { useSnackbar } from "../../contexts/snackbar";
import { useParams } from "react-router-dom";

interface NFTTileProps {
  data: MetadataData;
  cluster: string;
}

interface NFTTraits {
  trait_type: string;
  value: string;
}

interface URIData {
  name: string;
  description: string;
  image: string;
  attributes: NFTTraits[];
}

const NFTTile: React.FC<NFTTileProps> = ({ data, cluster }) => {
  const { theme } = useContext(ThemeContext);
  const { enqueueSnackbar } = useSnackbar();
  let { handle } = useParams<string>();

  const [nftSelect, toggleNftSelect] = useState(false);
  const [uRIData, setURIData] = useState<URIData>();

  const [editProfileMutation] = useEditProfileMutation({
    refetchQueries: [{ query: ProfileDocument, variables: { handle } }],
  });

  useMemo(() => {
    (async () => {
      try {
        const response = await fetch(data.data.uri);
        const json = await response.json();
        setURIData(json);
      } catch (error) {
        displayError(error, enqueueSnackbar);
      }
    })();
  }, []);

  const handleEditProfile = async (e: any) => {
    e.preventDefault();

    try {
      await editProfileMutation({
        variables: {
          handle,
          avatar: uRIData?.image,
          data: {
            avatarMint: data.mint,
            avatarUpdateAuthority: data.updateAuthority,
          },
        },
      });

      enqueueSnackbar("Your profile has been updated ðŸ¥³.", {
        variant: "success",
      });
    } catch (err) {
      return displayError(err, enqueueSnackbar);
    }
  };

  return (
    <>
      <Box
        pb={4}
        pr={4}
        sx={{
          display: "flex",
          flexDirection: "column",
          minWidth: "140px",
          cursor: "pointer",
        }}
        key={uRIData?.image}
      >
        <Link
          onClick={() => toggleNftSelect(true)}
          color={theme.secondaryColor}
          underline="hover"
        >
          <Box>
            {uRIData?.image ? (
              <img src={uRIData?.image} alt={uRIData?.name} width="120" />
            ) : (
              <DoNotDisturbOnIcon fontSize="large" />
            )}
          </Box>
          <Box>
            <Typography sx={{ fontSize: "0.8em", maxWidth: "120px" }}>
              {uRIData?.name}
            </Typography>
          </Box>
        </Link>
      </Box>
      <Modal
        open={nftSelect}
        onClose={() => toggleNftSelect(false)}
        aria-labelledby="NFT action select"
        aria-describedby="Apply action for given NFT"
      >
        <Box
          sx={{
            position: "absolute",
            top: "40%",
            left: "50%",
            transform: "translate(-50%, -50%)",
            padding: 2,
            bgcolor: theme.background,
            border: `2px solid ${theme.blue.lightest}`,
            borderRadius: 1,
            maxWidth: "25em",
            width: "95%",
          }}
        >
          <Box
            display="flex"
            justifyContent="space-between"
            alignItems="flex-start"
          >
            <Typography id="nft-modal-title" variant="h6" component="h2">
              {uRIData?.name}
            </Typography>
            <IconButton
              aria-label="close-modal"
              onClick={() => toggleNftSelect(false)}
              sx={{
                padding: 0,
              }}
            >
              <Clear fontSize="small" sx={{ color: theme.blue.lighter }} />
            </IconButton>
          </Box>
          <Box display="flex" mt={2}>
            <Stack spacing={2}>
              {uRIData?.image && (
                <img src={uRIData?.image} alt={uRIData?.name} width="120" />
              )}
              {/* Verified */}
            </Stack>
            <Box
              ml={1}
              display="flex"
              sx={{
                overflow: "hidden",
                flexWrap: "wrap",
                alignContent: "flex-start",
              }}
            >
              {uRIData?.attributes && typeof uRIData?.attributes?.map === 'function' &&
                uRIData.attributes.map((nftItem) => (
                  <Chip
                    label={`${nftItem.trait_type}: ${nftItem.value}`}
                    variant="outlined"
                    size="small"
                    sx={{
                      margin: "0 0.5em 0.5em 0",
                      fontSize: "0.7em",
                    }}
                  />
                ))}
              <Chip
                component="a"
                target="_blank"
                href={`https://solana.fm/address/${data?.mint}?cluster=${cluster}-solana`}
                label="View in explorer"
                variant="outlined"
                size="small"
                color="primary"
                sx={{
                  margin: "0 0.5em 0.5em 0",
                  cursor: "pointer",
                  fontSize: "0.7em",
                }}
              />
            </Box>
          </Box>
          <Stack mt={2} spacing={1}>
            <Button
              fullWidth={true}
              variant="contained"
              onClick={handleEditProfile}
            >
              Set Profile Picture
            </Button>
            <Button
              fullWidth={true}
              variant="outlined"
              onClick={() => toggleNftSelect(false)}
            >
              Cancel
            </Button>
          </Stack>
        </Box>
      </Modal>
    </>
  );
};

export const NFTGallery: React.FC<ProfileQuery> = ({ profile }) => {
  const { connection } = useConnection();
  const { name } = currentCluster();
  const { enqueueSnackbar } = useSnackbar();

  const [nfts, setNfts] = useState<MetadataData[]>();
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    (async () => {
      try {
        setLoading(true);
        const nftMeta = profile.publicAddress
          ? await Metadata.findDataByOwner(connection, profile.publicAddress)
          : [];
        setNfts(nftMeta);
      } catch (error) {
        displayError(error, enqueueSnackbar);
      } finally {
        setLoading(false);
      }
    })();
  }, []);

  if (loading) return <Loader />;

  return (
    <>
      {nfts?.length ? (
        <Stack
          direction="row"
          sx={{
            flexWrap: "wrap",
            justifyContent: "flex-start",
            alignItems: "baseline",
          }}
        >
          {nfts &&
            nfts.map((nft: MetadataData) => (
              <NFTTile data={nft} key={nft.data.uri} cluster={name} />
            ))}
        </Stack>
      ) : (
        <Box pt={10} sx={{ display: "flex", justifyContent: "center" }}>
          <Typography>No NFTs in your wallet</Typography>
        </Box>
      )}
    </>
  );
};
