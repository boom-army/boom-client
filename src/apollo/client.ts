import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          heroFeed: offsetLimitPagination(),
          getChannelById: offsetLimitPagination(),
          users: offsetLimitPagination(),
          searchTweets: offsetLimitPagination(),
          searchUser: offsetLimitPagination(),
          mentions: offsetLimitPagination(),
        },
      },
    },
  }),
  credentials: 'include',
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
    "Access-Control-Allow-Origin" : "*",
  },
  defaultOptions: {
    watchQuery: {
      context: async ({ req }: any) => {
        const token = req?.headers?.authorization || null;
        const user = token ? localStorage.getItem("user")?.toString() : null;
        return { user, isLoggedIn: !!token };
      },
    },
  },
});
