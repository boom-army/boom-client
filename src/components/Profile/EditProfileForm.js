import { useState } from 'react';
import Avatar from '@mui/material/Avatar';
import Button from '../../styles/Button';
import CoverPhoto from '../../styles/CoverPhoto';
import Form from '../../styles/Form';
import Input from '../Input';
import { TextareaAutosize } from "@mui/material";
import useInput from '../../hooks/useInput';
import { PROFILE, EDIT_PROFILE } from '../../queries/profile';
import { SIGN_FILE } from '../../queries/files';
import { displayError } from '../../utils';
import { uploadImage } from '../../utils';
import { useMutation } from '@apollo/client';
import { useSnackbar } from 'notistack';
import { useNavigate } from 'react-router-dom';

export const EditProfileForm = ({ profile, setUser }) => {
  const navigate = useNavigate();
  const [avatarState, setAvatar] = useState('');
  const [coverPhotoState, setCoverPhoto] = useState('');

  const handle = useInput(profile && profile.handle);
  const consumerName = useInput(profile && profile.consumerName);
  const location = useInput(profile && profile.location);
  const website = useInput(profile && profile.website);
  const dob = useInput(profile && profile.dob);
  const avatar = useInput(profile && profile.avatar);
  const bio = useInput(profile && profile.bio);
  const coverPhoto = useInput(profile && profile.coverPhoto);

  const [editProfileMutation, { loading }] = useMutation(EDIT_PROFILE, {
    refetchQueries: [{ query: PROFILE, variables: { handle: handle.value } }],
  });
  const [signFileMutation] = useMutation(SIGN_FILE);

  const { enqueueSnackbar } = useSnackbar();

  const handleEditProfile = async (e) => {
    e.preventDefault();

    if (!consumerName.value) {
      return enqueueSnackbar('You cannot leave name empty', {
        variant: 'error',
      });
    }

    try {
      const { data } = await editProfileMutation({
        variables: {
          handle: handle.value,
          consumerName: consumerName.value,
          dob: dob.value,
          bio: bio.value,
          location: location.value,
          website: website.value,
          avatar: avatarState ? avatarState : avatar.value,
          coverPhoto: coverPhotoState ? coverPhotoState : coverPhoto.value,
        },
      });

      setUser({
        handle: data.editProfile.handle,
        publicAddress: data.editProfile.publicAddress,
        id: data.editProfile.id,
      });

      navigate(`/${data.editProfile.handle}`);

      enqueueSnackbar('Your profile has been updated ðŸ¥³.', {
        variant: 'success',
      });
    } catch (err) {
      return displayError(err, enqueueSnackbar);
    }
  };

  const handleCoverPhoto = async (e) => {
    try {
      const file = e.target.files[0];
      const { data } = await signFileMutation({
        variables: {
          file: file.name,
          type: file.type,
        },
      });
      const signedUrl = data.signFileUrl;
      const imageData = await uploadImage(file, signedUrl, enqueueSnackbar);
      const imageUrl = imageData.config.url.split('?')[0];
      setCoverPhoto(imageUrl);
    } catch (error) {
      console.log(error);
    }
  };

  const handleAvatar = async (e) => {
    try {
      const file = e.target.files[0];
      const { data } = await signFileMutation({
        variables: {
          file: file.name,
          type: file.type,
        },
      });
      const signedUrl = data.signFileUrl;
      const imageData = await uploadImage(file, signedUrl, enqueueSnackbar);
      const imageUrl = imageData.config.url.split('?')[0];
      setAvatar(imageUrl);
    } catch (error) {
      console.log(error);
    }
  };

  return (
    <Form lg onSubmit={handleEditProfile}>
      <div className="cover-photo">
        <label htmlFor="cover-photo-input">
          <CoverPhoto
            src={coverPhotoState ? coverPhotoState : coverPhoto.value}
            alt="cover"
          />
        </label>
        <input
          type="file"
          id="cover-photo-input"
          accept="image/*"
          onChange={handleCoverPhoto}
        />
      </div>

      <div className="avatar-input">
        <label htmlFor="avatar-input-file">
          <Avatar src={avatarState ? avatarState : avatar.value} />
        </label>
        <input
          type="file"
          accept="image/*"
          id="avatar-input-file"
          onChange={handleAvatar}
        />
      </div>

      <Input
        lg={true}
        text="Handle"
        value={handle.value}
        onChange={handle.onChange}
      />

      <Input
        lg={true}
        text="Name"
        value={consumerName.value}
        onChange={consumerName.onChange}
      />
      <div className="bio-wrapper">
        <label className="bio" htmlFor="bio">
          Bio
        </label>
        <TextareaAutosize
          id="bio"
          placeholder="Bio"
          value={bio.value}
          onChange={bio.onChange}
        />
      </div>
      <Input
        lg={true}
        text="Website"
        value={website.value}
        onChange={website.onChange}
      />
      <Input
        lg={true}
        text="Date of Birth"
        value={dob.value}
        onChange={dob.onChange}
      />
      <Input
        lg={true}
        text="Location"
        value={location.value}
        onChange={location.onChange}
      />
      <Button outline disabled={loading} type="submit">
        {loading ? 'Saving' : 'Save'}
      </Button>
    </Form>
  );
};
