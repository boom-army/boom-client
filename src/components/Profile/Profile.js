import React from "react";
import { useQuery } from '@apollo/client';
import { useParams } from "react-router-dom";
import styled from "styled-components";
import ProfileInfo from "./ProfileInfo";
import { ShowTweet } from "../Tweet";
import { Loader } from "../Loader";
import { PROFILE } from "../../queries/profile";

const Wrapper = styled.div`
	padding-bottom: 5rem;

  .profile-top {
    display: flex;
    flex-direction: column;
    margin-left: 1rem;

    span.tweetsCount {
      margin-top: 0.1rem;
      color: ${(props) => props.theme.secondaryColor};
      font-size: 0.9rem;
    }
  }
`;

export const Profile = () => {
  const { handle } = useParams();

  const { loading, error, data } = useQuery(PROFILE, {
    variables: { handle },
  });

  if (loading) return <Loader />;

  return (
    <Wrapper>
      {/* <div className="profile-top">
        <span>{data && data.profile && data.profile.consumerName}</span>
        <span className="tweetsCount">
          {data && data.profile && data.profile.tweetsCount
            ? `${data.profile.tweetsCount} Tweets`
            : "No Tweets"}
        </span>
      </div> */}
      <ProfileInfo profile={data && data.profile} />
      {data && data.profile && data.profile.tweets && data.profile.tweets.length
        ? data.profile.tweets.map((tweet) => (
          <ShowTweet key={tweet.id} tweet={tweet} />
        ))
        : null}
    </Wrapper>
  );
};
