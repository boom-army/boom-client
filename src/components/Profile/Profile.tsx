import React, { useEffect, useState } from "react";
import ProfileInfo from "./ProfileInfo";
import { Box, Tab } from "@mui/material";
import { Loader } from "../Loader";
import { Meeps } from "./Meeps";
import { TabContext, TabPanel, TabList } from "@mui/lab";
import { useParams } from "react-router-dom";
import { useProfileLazyQuery } from "../../generated/graphql";
import { NFTGallery } from "./NFTGallery";
import { CustomResponse } from "../CustomResponse";

export const Profile: React.FC = () => {
  const [tabValue, setTabValue] = useState("1");

  let { handle } = useParams<string>();

  const [setHandle, { loading, data }] = useProfileLazyQuery();

  useEffect(() => {
    if (handle) setHandle({ variables: { handle } });
  }, [handle, setHandle]);

  const handleChange = (event: React.SyntheticEvent, newValue: string) => {
    setTabValue(newValue);
  };

  if (loading) {
    return <Loader />;
  }
  if (!loading && !data) {
    return (
      <CustomResponse text="Oops, you are trying to visit a profile which doesn't exist. Make sure the profile handle exists" />
    );
  }

  return (
    <>
      <ProfileInfo profile={data && data.profile} />
      <Box sx={{ typography: "body1" }}>
        <TabContext value={tabValue}>
          <Box sx={{ borderBottom: 2, borderColor: "#374148" }}>
            <TabList
              onChange={handleChange}
              aria-label="Profile tablist select"
            >
              <Tab sx={{ minWidth: 150 }} label="NFT Gallery" value="1" />
              <Tab label="Meeps" value="2" />
            </TabList>
          </Box>
          <TabPanel value="1">
            {data && <NFTGallery publicAddress={data?.profile.publicAddress} />}
          </TabPanel>
          <TabPanel value="2">
            <Meeps data={data} />
          </TabPanel>
        </TabContext>
      </Box>
    </>
  );
};
