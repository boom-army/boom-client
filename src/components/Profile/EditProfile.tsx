import { Loader } from "../Loader";
import { EditProfileForm } from "./EditProfileForm";
import { styled } from "@mui/material/styles";

const Wrapper = styled("div")((props) => ({
  paddingBottom: "5rem",

  ".flex-wrapper": {
    display: "flex",
    justifyContent: "center",

    form: {
      ".cover-photo": {
        marginBottom: "1rem",
        cursor: "pointer",
      },

      ".avatar-input": {
        marginTop: "-100px",
        marginLeft: "1rem",
        cursor: "pointer",
      },

      "div.bio-wrapper": {
        background: props.theme.tertiaryColor2,
        marginBottom: "1.4rem",
        borderBottom: `1px solid ${props.theme.accentColor}`,
        padding: "0.5rem",

        label: {
          color: props.theme.secondaryColor,
          marginBottom: "0.4rem",
        },

        textarea: {
          fontSize: "1rem",
          width: "100%",
          background: props.theme.tertiaryColor2,
          border: "none",
          fontFamily: props.theme.font,
          color: props.theme.primaryColor,
        },
      },
    },
  },
  "@media screen and (max-width: 760px)": {
    ".flex-wrapper": {
      display: "block",
    },
  },
}));

export const EditProfile = ({ data, loading, setUser }: any) => {
  if (loading) return <Loader />;

  return (
    <Wrapper>
      <div className="flex-wrapper">
        <EditProfileForm profile={data && data?.profile} setUser={setUser} />
      </div>
    </Wrapper>
  );
};
