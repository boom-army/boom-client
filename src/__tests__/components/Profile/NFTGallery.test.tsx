import { ProfileDocument } from "../../../generated/graphql";
import React from "react";
import { NFTGallery } from "../../../components/Profile/NFTGallery";
import { BrowserRouter } from "react-router-dom";
import { MockedProvider } from "@apollo/client/testing";
import userEvent from "@testing-library/user-event";
import { MemoryRouter } from "react-router-dom";
import { act, fireEvent, render } from "@testing-library/react";
import { SnackbarProvider } from "notistack";
import {
  useEditProfileMutation,
  EditProfileDocument,
} from "../../../generated/graphql";


export const __mocks__: any = [
  {
    request: {
    //   query: EditProfileDocument,
      variables: {
       
      },
    },
    result: {
      data: {
        // editProfile: {
        //   id: "ckzny1iv10031lmn568gj3b3n",
        //   handle: "shy-cloud-4965",
        //   publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
        //   __typename: "User",
        // },
      },
      // errors: "An error occurred",
    },
  },
];



test("NFT Gallery ", async () => {
  let rendered;
  await act(async () => {
    rendered = render(
    //   <MockedProvider mocks={__mocks__} addTypename={false}>
    <SnackbarProvider>
       <NFTGallery publicAddress="JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4" />
       </SnackbarProvider>
    //   </MockedProvider>,{wrapper: MemoryRouter}
    );
  });
  expect(rendered).toMatchSnapshot();
});


