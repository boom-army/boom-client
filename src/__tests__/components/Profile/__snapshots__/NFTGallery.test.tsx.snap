// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NFT Gallery  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-uz95m6"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          No NFTs in your wallet
        </p>
      </div>
      <div
        class="makeStyles-bottom-4 makeStyles-left-5 makeStyles-root-1"
      >
        <div
          class="MuiCollapse-root MuiCollapse-entered"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                class="SnackbarItem-root-8 SnackbarItem-wrappedRoot-23 SnackbarItem-anchorOriginBottomLeft-14"
              >
                <div
                  aria-describedby="notistack-snackbar"
                  class="ForwardRef-root-24 SnackbarItem-contentRoot-15 SnackbarItem-variantError-18 SnackbarItem-lessPadding-16"
                  role="alert"
                  style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
                >
                  <div
                    class="SnackbarItem-message-21"
                    id="notistack-snackbar"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      style="font-size: 20px; margin-inline-end: 8px;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12,2C17.53,2 22,6.47 22,12C22,17.53 17.53,22 12,22C6.47,22 2,17.53 2,12C2,
        6.47 6.47,2 12,2M15.59,7L12,10.59L8.41,7L7,8.41L10.59,12L7,15.59L8.41,17L12,
        13.41L15.59,17L17,15.59L13.41,12L17,8.41L15.59,7Z"
                      />
                    </svg>
                    Cannot read property 'getTokenAccountsByOwner' of undefined
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-uz95m6"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        No NFTs in your wallet
      </p>
    </div>
    <div
      class="makeStyles-bottom-4 makeStyles-left-5 makeStyles-root-1"
    >
      <div
        class="MuiCollapse-root MuiCollapse-entered"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <div
              class="SnackbarItem-root-8 SnackbarItem-wrappedRoot-23 SnackbarItem-anchorOriginBottomLeft-14"
            >
              <div
                aria-describedby="notistack-snackbar"
                class="ForwardRef-root-24 SnackbarItem-contentRoot-15 SnackbarItem-variantError-18 SnackbarItem-lessPadding-16"
                role="alert"
                style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
              >
                <div
                  class="SnackbarItem-message-21"
                  id="notistack-snackbar"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    style="font-size: 20px; margin-inline-end: 8px;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12,2C17.53,2 22,6.47 22,12C22,17.53 17.53,22 12,22C6.47,22 2,17.53 2,12C2,
        6.47 6.47,2 12,2M15.59,7L12,10.59L8.41,7L7,8.41L10.59,12L7,15.59L8.41,17L12,
        13.41L15.59,17L17,15.59L13.41,12L17,8.41L15.59,7Z"
                    />
                  </svg>
                  Cannot read property 'getTokenAccountsByOwner' of undefined
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
